cirq-core/cirq/work/observable_measurement.py

```diff
@@ -354,6 +355,47 @@ def _to_sweep(param_tuples):
     return to_sweep
 
 
+def _parse_checkpoint_options(
+    checkpoint: bool, checkpoint_fn: Optional[str], checkpoint_other_fn: Optional[str]
+) -> Tuple[Optional[str], Optional[str]]:
+    """Parse the checkpoint-oriented options in `measure_grouped_settings`.
+
+    This function contains the validation and defaults logic. Please see
+    `measure_grouped_settings` for documentation on these args.
+
+    Returns:
+        checkpoint_fn, checkpoint_other_fn: Parsed or default filenames for primary and previous
+            checkpoint files.
+    """
+    if not checkpoint:
+        if checkpoint_fn is not None or checkpoint_other_fn is not None:
+            raise ValueError(
+                "Checkpoint filenames were provided but `checkpoint` was set to False."
+            )
+        return None, None
+
+    if checkpoint_fn is None:
+        checkpoint_dir = tempfile.mkdtemp()
```

Comment:
you're right. I added the pytest `tmpdir` fixture to the test. For the actual code, we want the checkpoint files to stick around

Reply:
Would `os.mkdir` be preferable, given that the files are meant to outlive the Python process (i.e. they are non-temporary)?

Reply:
`mkdir` makes a directory but you'd still need to choose its location. This will give you a file in `/tmp` which gets cleaned up after like 30 days or whatever your computer's policy is.